### PR TITLE
Allow poll interval to be specified from the command line.

### LINKED ISSUE #46: Allow server poll time to be configured as a command line option
Useful for performance tuning.

### PR DESCRIPTION
**Pull request by fonglh**
Uses 10 seconds as the default.

Fixes #46.

### CODE REVIEW

File: lib/coursemology/evaluator/client.rb [hunk 1]
```diff
@@ -31,7 +35,7 @@ def client_loop
       # :nocov:
       # This sleep might not be triggered in the specs, because interruptions to the thread is
       # nondeterministically run by the OS scheduler.
-      sleep(1.minute)
+      sleep(@poll_time.send(@poll_interval))
```
Comment (lowjoel) [inline lib/coursemology/evaluator/client.rb]: This is very susceptible to injection.

File: lib/coursemology/evaluator/client.rb [hunk 2]
```diff
@@ -9,7 +9,11 @@ def self.initialize(host, api_user_email, api_token)
   end
 
   # @param [Boolean] one_shot If the client should only fire one request.
-  def initialize(one_shot = false)
+  # @param [Float] poll_time Sleep time between checks for task allocation.
+  # @param [String] poll_interval Units for poll_time.
+  def initialize(one_shot = false, poll_time = 10, poll_interval = 'seconds')
```
Comment (lowjoel) [inline lib/coursemology/evaluator/client.rb]: I prefer a string with the unit attached and parsing be done. "10s" or "00:10" or the like. There should be ActiveSupport helpers for that.

File: lib/coursemology/evaluator/client.rb (hunk 1 above)
Comment (lowjoel) [inline lib/coursemology/evaluator/client.rb]: Also, I'm not sure if poll intervals are being tested

Comment (fonglh) [inline lib/coursemology/evaluator/client.rb]: No they aren't, I tested this manually.

File: lib/coursemology/evaluator/client.rb (hunk 2 above)
Comment (fonglh) [inline lib/coursemology/evaluator/client.rb]: Ok I'll check it out.

File: lib/coursemology/evaluator/client.rb (hunk 1 above)
Comment (lowjoel) [inline lib/coursemology/evaluator/client.rb]: You can stub out sleep, and check that the correct timeout is specified as a spec

File: lib/coursemology/evaluator/client.rb (hunk 2 above)
Comment (fonglh) [inline lib/coursemology/evaluator/client.rb]: It looks like a Rails 5 addition.
http://api.rubyonrails.org/classes/ActiveSupport/Duration.html#method-c-parse

**Comment by fonglh** [comment]
@allenwq suggested just forcing the units to be in seconds. That'll remove the need for parsing and removes the dependency on the ISO8601 gem. The number can also be passed directly to the sleep function.

**Comment by allenwq** [comment]
But since you have done it already, it's OK to stick with the current way.  Just that I feel using seconds is enough.

**Comment by fonglh** [comment]
I just remembered I also want to do customizable Docker image cache interval. That could vary over a much greater time frame so I'll still need the ISO8601 gem.

**Comment by allenwq** [comment]
LGTM

**Comment by allenwq** [comment]
I will merge this first since @fonglh need it for deploy.

**Pull request merged**